### PR TITLE
chore: add e2e tests for notifications sms & in-app-messaging

### PR DESCRIPTION
**Pull request by awsluja**
#### Description of changes
This adds test coverage for pre-existing notification channels (SMS), and ensure compatibility between those channels and the new channel (In-App Messaging). It also make sure that both SMS/In-App Messaging channels continue to work well with Pinpoint/Analytics.

These test cases ensure the following flows work, and that configuration files are set correctly.

1. adding/removing notifications in a new project (SMS)
2. adding/removing notifications in a new project (in-app messaging)
3. adding/removing notifications in a project that has pinpoint configured already (SMS)
4. adding/removing notifications in a project that has pinpoint configured already (in-app messaging)

Additionally, several checks for file configuration states, locally & in the cloud, are included.

See slack for information on test cases.

E2E is running at: https://app.circleci.com/pipelines/github/aws-amplify/amplify-cli?branch=run-e2e%2Faluja%2Fadd-e2e-notifications-in-app

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies
- [ ] [Pull request labels](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#labels) are added

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-e2e-tests/src/__tests__/notifications-analytics-compatibility-in-app-2.test.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    await removeAnalytics(projectRoot, {});
+    await amplifyPushUpdate(projectRoot);
+
+    await amplifyStatus(projectRoot, 'Auth');
+
+    // notification should not exist in the cloud
+    // TBD: at this time, removing pinpoint doesn't remove notifications, but we intend to
+    // re-enable this check once updated
+    // const endCloudBackendMeta = await getProjectMeta(projectRoot);
+    // expect(endCloudBackendMeta.notifications).toBeUndefined();
```
Comment (edwardfoyle) [inline packages/amplify-e2e-tests/src/__tests__/notifications-analytics-compatibility-in-app-2.test.ts:80]: should this be removed?

File: packages/amplify-e2e-tests/src/__tests__/notifications-analytics-compatibility-sms-1.test.ts [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+  afterEach(async () => {
+    await removeAnalytics(projectRoot, {});
+    await deleteProject(projectRoot);
+    deleteProjectDir(projectRoot);
+    if (pullTestProjectRoot) {
+      deleteProjectDir(pullTestProjectRoot);
+    }
+  });
+
+  it(`should work well with pre-existing pinpoint that hasn't pushed`, async () => {
```
Comment (edwardfoyle) [inline packages/amplify-e2e-tests/src/__tests__/notifications-analytics-compatibility-sms-1.test.ts:52]: nit:
```suggestion
  it(`should work with pre-existing pinpoint that hasn't pushed`, async () => {
```

File: packages/amplify-e2e-tests/src/__tests__/notifications-analytics-compatibility-sms-2.test.ts [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+  afterEach(async () => {
+    await removeAnalytics(projectRoot, {});
+    await deleteProject(projectRoot);
+    deleteProjectDir(projectRoot);
+    if (pullTestProjectRoot) {
+      deleteProjectDir(pullTestProjectRoot);
+    }
+  });
+
+  it(`should work well with pre-existing pinpoint that has pushed`, async () => {
```
Comment (edwardfoyle) [inline packages/amplify-e2e-tests/src/__tests__/notifications-analytics-compatibility-sms-2.test.ts:53]:
```suggestion
  it(`should work with pre-existing pinpoint that has pushed`, async () => {
```

File: packages/amplify-e2e-tests/src/__tests__/notifications-fcm.test.ts [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    projectRoot = await createNewProjectDir(projectPrefix);
+  });
+
+  afterEach(async () => {
+    await deleteProject(projectRoot);
+    deleteProjectDir(projectRoot);
+  });
+
+  // Skipping FCM for now until we have setup required e2e accounts
+  it.skip(`should add and remove the ${testChannel} channel correctly when no pinpoint is configured`, async () => {
```
Comment (edwardfoyle) [inline packages/amplify-e2e-tests/src/__tests__/notifications-fcm.test.ts:37]: Would these tests be better as a [`test.each`](https://jestjs.io/docs/api#testeachtablename-fn-timeout) block? Looks like they are somewhat different so maybe not a good fit, just something to think about.

**Review by edwardfoyle — APPROVED** [review]

File: packages/amplify-e2e-tests/src/__tests__/notifications-analytics-compatibility-in-app-2.test.ts (hunk 1 above)
Comment (awsluja) [reply packages/amplify-e2e-tests/src/__tests__/notifications-analytics-compatibility-in-app-2.test.ts:80]: ah, thats right

File: packages/amplify-e2e-tests/src/__tests__/notifications-fcm.test.ts (hunk 4 above)
Comment (awsluja) [reply packages/amplify-e2e-tests/src/__tests__/notifications-fcm.test.ts:37]: we're not entirely sure how these tests will end up looking after we add the supporting e2e accounts; but will keep an eye on this

File: packages/amplify-category-notifications/src/commands/notifications/remove.ts [hunk 5]
```diff
@@ -71,13 +71,15 @@ export const run = async (context: $TSContext): Promise<$TSContext> => {
...
         await writeData(context, channelAPIResponse);
+        printer.info('The channel has been successfully disabled.');
       }
     } else if (isPinpointAppOwnedByNotifications(pinpointAppStatus.status)) {
       const confirmDelete = await prompter.confirmContinue('Confirm that you want to delete the associated Amazon Pinpoint application');
       if (confirmDelete) {
         await deletePinpointApp(context);
         printer.info('The Pinpoint application has been successfully deleted.');
         await writeData(context, undefined);
+        printer.info('The Pinpoint application has been successfully deleted.');
```
[the remark this answers is not in the document]
Comment (awsluja) [reply packages/amplify-category-notifications/src/commands/notifications/remove.ts:81]: good catch, keeping the last line only - PR updated just now

**Review by sobolk — APPROVED** [review]
image

**Review by jhockett — APPROVED** [review]

**Pull request merged**